src/Gd/Layers.php

```diff
@@ -50,7 +50,7 @@ class Layers extends AbstractLayers
      */
     public function __construct(Image $image, PaletteInterface $palette, $resource, $initialOffset = 0)
     {
-        if (!is_resource($resource)) {
+        if (!$resource instanceof \GdImage && !is_resource($resource)) {
```

Comment:
The `@param` annotation for `$resource` should be adjusted as well:

```
@param resource|\GdImage $resource
```

Reply:
Fixed in e424a69a354dd46a57467a841128bad10bc249e2